bundles/org.openhab.ui/web/src/pages/settings/things/link/link-edit.vue

```diff
@@ -123,9 +122,10 @@ export default {
       const itemName = this.item.name
       const itemType = this.item.type
       const channelUID = this.channel.uid.replace('#', '%23')
-      const getProfileTypes = this.$oh.api.get('/rest/profile-types?channelTypeUID=' + this.channel.channelTypeUID + '&itemType=' + itemType)
-      getProfileTypes.then((data) => {
+      this.$oh.api.get('/rest/profile-types?channelTypeUID=' + this.channel.channelTypeUID + '&itemType=' + itemType).then((data) => {
         this.profileTypes = data
+        this.profileTypes.unshift(data.splice(data.findIndex(p => p.uid === 'system:default'), 1)[0]) // move default to be first
+          .filter(p => !p.supportedItemTypes.length || p.supportedItemTypes.includes(this.item.type)) // only show compatible profile types
```

Comment:
I get an error here:

_Uncaught (in promise) TypeError: this.profileTypes.unshift(...).filter is not a function
    at eval (link-edit.vue?c1b6:128)_

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift

> The unshift() method adds one or more elements to the beginning of an array and returns **the new length of the array.**

Reply:
oh wow! not sure how that made it into the commit, possibly happened during rebasing. Will correct.